Changed architecture

jottyVlad opened the pull request:
Rewrited configs and APIs registration, added ISocialNet interface

jottyVlad committed:
Fixed naming in sender.py

jottyVlad committed:
Added VK support

jottyVlad committed:
Merge branch 'master' into headers_classes

jottyVlad committed:
Merge pull request #5 from jottyVlad/headers_classes

Close headers_classes branch

jottyVlad committed:
PEP8 fix

jottyVlad committed:
Rewrited APIs registration, social net registration

jottyVlad committed:
Init SocialNet without calling init_net method

jottyVlad committed:
Changed config structure

jottyVlad committed:
Changed interfaces structure; added ISocialNet interface

kiriharu commented on api/social_nets/interfaces.py:
```diff
@@ -0,0 +1,32 @@
+
+    @property
+    @abstractmethod
+    def api_url(self):
```
Maybe use @abstractproperty?

jottyVlad replied:
> Maybe use @abstractproperty?

No, because @abstractproperty now is deprecated and we should use @property with @abstractmethod

kiriharu commented on api/service_apis/covid19.py:
```diff
@@ -34,14 +40,14 @@ def __str__(self) -> str:
 
 class Covid19(IApi, ABC):
```
Is ABC really needed here?

jottyVlad replied:
No, I replaced it.

kiriharu commented on api/service_apis/qiwi.py:
```diff
@@ -29,12 +35,12 @@ def __str__(self) -> str:
 class Qiwi(IApi, ABC):
```
Same for ABC

jottyVlad replied:
Did it.

kiriharu commented on api/service_apis/wttr_in.py:
```diff
@@ -31,9 +36,9 @@ def __str__(self) -> str:
 
 
 class WttrIn(IApi, ABC):
```
Same for ABC

jottyVlad replied:
Did it

kiriharu commented on service/registration.py:
```diff
@@ -0,0 +1,70 @@
+            self.net = Vkontakte(token)
+
+        elif self.type_net == SocialNetType.Discord:
+            pass
```
Maybe raise NotImpimented?

jottyVlad replied:
> Maybe raise NotImpimented?

Yes, I did it.

kiriharu reviewed (changes requested):
Great PR, but I have several questions for you:
1) Can you explain, why you using ABC in class implementations (like Telegram). I think you can just Inherit base class.
2) You added a few new social networks. Can you update readme?

jottyVlad committed:
Fixed: classes of social networks and services were abstract.
Now when you want to define discord social network, will be raised NotImplemented.
Changed README

jottyVlad committed:
Fixed: incorrect VK API version

kiriharu commented:
Thank you!
